vehicle/mkv/software/dashboard/dashboard.c

```diff
@@ -1,521 +1,144 @@
-				PORT_RJ_LED2 |= _BV(RJ_LED2);
-			} else{
-				PORT_RJ_LED2 &= ~_BV(RJ_LED2);
-			}
-
-			if(bit_is_clear(gFlag, PRECHARGE) && bit_is_set(gFlag, BRAKE_PRESSED)) {
-				PORT_START_LED |= _BV(START_LED);
-			} else{
-				PORT_START_LED &= ~_BV(START_LED);
-			}
-
-			if(bit_is_set(gFlag, STATUS_START)) {
-				gCAN_MSG[2] = 0xFF;
-			} else{
-				gCAN_MSG[2] = 0x00;
-			}
-		}
-	}
+int main(void) {
+    // Initialization
+    can_init(BAUD_500KBPS);
+    timer_init(&timer0_cfg);
+    adc_init();
+
+    // Set pin modes
+    gpio_set_mode(IMD_LED, OUTPUT);
+    gpio_set_mode(AMS_LED, OUTPUT);
+    gpio_set_mode(START_LED, OUTPUT);
+    gpio_set_mode(HV_LED, OUTPUT);
+    gpio_set_mode(LV_LED, OUTPUT);
+    gpio_set_mode(BRAKE_LED, OUTPUT);
+    gpio_set_mode(RTD_BUZZER_LSD, OUTPUT);
+
+    gpio_set_mode(START_BTN, INPUT);
+
+    // Enable interrupts
+    sei();
+
+    // Turn on LV LED
+    gpio_set_pin(LV_LED);
+
+    uint16_t steering_pos;
+
+    for (;;) {
+        steering_pos = adc_read(STEERING_POS);
+
+        // Brakelight message for Brakelight LED and
+        // check
+        if (can_receive_brakelight_bspd() == 0) {
+            if (brakelight_bspd.brake_gate == 0xFF) {
+                BRAKE_PRESSED = true;
+                gpio_set_pin(BRAKE_LED);
+            } else {
+                BRAKE_PRESSED = false;
+                gpio_clear_pin(BRAKE_LED);
+            }
+        }
+
+        // BMS Core message for BMS Status LED
+        if (can_receive_bms_core() == 0) {
+            if (bms_core.bms_ok == true) { // check BMS status
+                gpio_set_pin(AMS_LED); // set BMS light high
+            } else {
+                gpio_clear_pin(AMS_LED); // clear BMS light
+            }
+        }
+
+        // AIR Control Critical message for HV LED and disabling ReadyToDrive if
+        // HV goes down
+        if (can_receive_air_control_critical() == 0) {
+            if (air_control_critical.air_p_status == false
+                && air_control_critical.air_n_status == false) {
+                HV_STATE = true;
+                gpio_set_pin(HV_LED); // set HV LED
+            } else {
+                HV_STATE = false;
+                READYTODRIVE = false; // Disable RTD
+                gpio_clear_pin(HV_LED); // clear HV LED
+                buzzer_counter = 0; // reset counter for next RTD cycle
+            }
+        }
+
+        // Is this now under Air Control Critical?
+        if (can_receive_air_control_critical() == 0) {
+            if (air_control_critical.imd_status == false) {
+                gpio_set_pin(IMD_LED); // set IMD light high
+            } else {
+                gpio_clear_pin(IMD_LED); // set IMD light low
+            }
+        }
+
+        // Throttle message for interfacing with LED Bars Board - TODO (SPI
+        // library)
+        if (can_receive_throttle() == 0) {
+            // do some stuff here
+        }
+
+        if (send_can) {
+            dashboard.fault_code = 0;
```

Comment:
You should set these variables elsewhere, not in the `send_can` block. For `ready_to_drive`, for example, you can set that variable in the interrupt where the `READYTODRIVE` variable is set.